src/integrationtest/java/infrastructure/TcpListener.java

```diff
@@ -77,7 +77,9 @@ public void run(){
     }
 
     public ServerSocket createSocket() throws IOException {
-        int[] ports = { 3843,4584, 4843, 60000 };
+        //int[] ports = { 3843,4584, 4843, 60000 };
```

Comment:
What happens if port 8080 is not available. If I remember correctly, this is a possibility and therefore that's why we have a list of ports.

Reply:
adfs on prem send "unregistered redirect url" fo such urls.
so aad accept redirect url with any port but not adfs.

Reply:
This is also the case for B2C. In that case we should just ask the lab to register redirect URLs with these ports (as they had done previously for B2C). Less of a chance of tests failing due to port being unavailable (which was a problem when I tried just using one port for all tests)

Reply:
we can just wait if port is busy, and try few times.

Reply:
My preference would be to use multiple ports, as that was seems more resilient and has been working perfectly. If you think doing retries is preferable, that's fine with me as well. In that case, you'd need to add the retry logic, as right now if the port is not available it fails and moves on.

Reply:
I have added retry logic, will ask lab teams to add few more ports, so we can do both